tests/integration/models/multifidelity/test_multifidelity_models.py

```diff
@@ -0,0 +1,574 @@
+
+    test_locations = tf.Variable(np.linspace(0, 10, 32), dtype=tf.float64)[:, None]
+    hf_test_locations = add_fidelity_column(test_locations, 1)
+
+    hf_samples = model.sample(hf_test_locations, 1_000_000)
+    hf_samples_direct = model.fidelity_residual_models[1].sample(test_locations, 1_000_000)
+
+    hf_samples_mean = tf.reduce_mean(hf_samples, axis=0)
+    hf_samples_var = tf.math.reduce_variance(hf_samples, axis=0)
+    hf_samples_direct_mean = tf.reduce_mean(hf_samples_direct, axis=0)
+    hf_samples_direct_var = tf.math.reduce_variance(hf_samples_direct, axis=0)
+
+    npt.assert_allclose(hf_samples_mean, hf_samples_direct_mean, atol=1e-3)
+    npt.assert_allclose(hf_samples_var, hf_samples_direct_var, atol=1e-3)
+
+
+def test_multifidelity_autoregressive_sample_hf_is_consistent_when_lf_is_flat() -> None:
+
+    xs_low = tf.Variable(np.linspace(0, 10, 100), dtype=tf.float64)[:, None]
+    xs_high = tf.Variable(np.linspace(0, 10, 10), dtype=tf.float64)[:, None]
+    lf_obs = tf.sin(xs_low)
+    hf_obs = 2 * tf.sin(xs_high) + tf.random.normal(
+        xs_high.shape, mean=0, stddev=1e-1, dtype=tf.float64
+    )
+
+    lf_query_points = add_fidelity_column(xs_low, 0)
+    hf_query_points = add_fidelity_column(xs_high, 1)
+
+    lf_dataset = Dataset(lf_query_points, lf_obs)
+    hf_dataset = Dataset(hf_query_points, hf_obs)
+
+    dataset = lf_dataset + hf_dataset
+
+    search_space = Box([0.0], [10.0])
+
+    model = MultifidelityAutoregressive(
+        build_multifidelity_autoregressive_models(
+            dataset, num_fidelities=2, input_search_space=search_space
+        )
+    )
+
+    model.update(dataset)
+    model.optimize(dataset)
+
+    kernel = gpflow.kernels.Matern52(variance=1e-18, lengthscales=[0.2, 0.2])
+    gpr = gpflow.models.GPR(dataset.astuple(), kernel, noise_variance=1e-5)
+
+    model.lowest_fidelity_signal_model = GaussianProcessRegression(gpr)
+
+    # Add some low fidelity points to check that predict on different fids works
+    test_locations_30 = tf.Variable(np.linspace(0, 10, 30), dtype=tf.float64)[:, None]
+    lf_test_locations = add_fidelity_column(test_locations_30, 0)
+    test_locations_32 = tf.Variable(np.linspace(0, 10, 32), dtype=tf.float64)[:, None]
+    hf_test_locations = add_fidelity_column(test_locations_32, 1)
+
+    concatenated_test_locations = tf.concat([lf_test_locations, hf_test_locations], axis=0)
+
+    concat_samples = model.sample(concatenated_test_locations, 1_000_000)
+    hf_samples = concat_samples[:, 30:]
+    hf_samples_direct = model.fidelity_residual_models[1].sample(test_locations_32, 1_000_000)
+
+    hf_samples_mean = tf.reduce_mean(hf_samples, axis=0)
+    hf_samples_var = tf.math.reduce_variance(hf_samples, axis=0)
+    hf_samples_direct_mean = tf.reduce_mean(hf_samples_direct, axis=0)
+    hf_samples_direct_var = tf.math.reduce_variance(hf_samples_direct, axis=0)
+
+    npt.assert_allclose(hf_samples_mean, hf_samples_direct_mean, atol=1e-3)
```

Comment:
maybe the way the LF model is "flattened" creates a little bit of noise, if we do it in a different way we might be able to tighten this tolerance? at least for the mean with 1e6 samples it seems a bit high...